python_modules/dagster/dagster/core/execution/context/input.py

```diff
@@ -231,7 +231,10 @@ def partition_key(self) -> str:
 
     @property
     def has_asset_partitions(self) -> bool:
-        return self.step_context.has_asset_partitions_for_input(self.name)
+        if self._step_context is not None:
+            return self._step_context.has_asset_partitions_for_input(self.name)
+        else:
```

Comment:
nit: 

useless "else"

Reply:
I'm not super opinionated, but my general take is that having an `else` in this kind of situation is clearer: explicitly cleaves the space of possibilities into two disjoint sets.

Maybe it's worth doing some project-level alignment on a convention for this.